osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -0,0 +1,207 @@
+  case NO_INHERITANCE:
+    return "No Inheritence";
+  default:
+    return "Other";
+  }
+}
+
+// helper function to get account/group name from trustee
+std::string trusteeToStr(const TRUSTEE& t) {
+  char name[256];
+  char domain[256];
+  unsigned long size = 256; // Max username length for Windows
+  SID_NAME_USE accountType;
+
+  switch (t.TrusteeForm) {
+  case TRUSTEE_IS_SID:
+    // get the name from the SID
+    if (!LookupAccountSid(nullptr,
+                          reinterpret_cast<PSID>(t.ptstrName),
+                          name,
+                          &size,
+                          domain,
+                          &size,
+                          &accountType)) {
+      TLOG << "LookupAccountSid error: " << GetLastError();
+    } else {
+      return name;
+    }
+  case TRUSTEE_IS_NAME:
+    // get the name from ptstrName
+    return t.ptstrName;
+  case TRUSTEE_BAD_FORM:
+    // invalid
+    return "Invalid";
+  case TRUSTEE_IS_OBJECTS_AND_SID:
+    // ptstrName member is a pointer to an OBJECTS_AND_SID struct
+    if (!LookupAccountSid(
+            nullptr,
+            reinterpret_cast<PSID>(
+                reinterpret_cast<OBJECTS_AND_SID*>(t.ptstrName)->pSid),
+            name,
+            &size,
+            domain,
+            &size,
+            &accountType)) {
+      // printf("LookupAccountSid error: %u\n", GetLastError());
+      TLOG << "LookupAccountSid error: " << GetLastError();
+    } else {
+      return name;
+    }
+  case TRUSTEE_IS_OBJECTS_AND_NAME:
+    // ptstrName member is a pointer to an OBJECTS_AND_NAME struct
+    return reinterpret_cast<OBJECTS_AND_NAME_*>(t.ptstrName)->ptstrName;
+  default:
+    return "";
+  }
+}
+
+QueryData genNTFSACLPerms(QueryContext& context) {
+  QueryData results;
+  unsigned long result = 0;
+  PACL DACL = nullptr;
+  PEXPLICIT_ACCESS aceList = nullptr, aceItem = nullptr;
+  unsigned long aceCount = 0, aceIndex = 0;
+
+  auto paths = context.constraints["path"].getAll(EQUALS);
+  for (const auto& path_string : paths) {
+    if (fs::exists(path_string)) {
+      // Get a pointer to the existing DACL.
+
+      result = GetNamedSecurityInfo(path_string.c_str(),
+                                    SE_FILE_OBJECT,
+                                    DACL_SECURITY_INFORMATION,
+                                    NULL,
+                                    NULL,
+                                    &DACL,
+                                    NULL,
+                                    NULL);
+      if (ERROR_SUCCESS != result) {
+        TLOG << "GetExplicitEnteriesFromAcl Error " << result;
+      }
+
+      // get list of ACEs from DACL pointer
+      result = GetExplicitEntriesFromAcl(DACL, &aceCount, &aceList);
+      if (ERROR_SUCCESS != result) {
+        TLOG << "GetExplicitEnteriesFromAcl Error " << result;
+      }
+
+      // Loop through list of entries
+
+      aceItem = aceList;
+
+      for (aceIndex = 0; aceIndex < aceCount; aceItem++) {
```

Comment:
Given that this is a `for` loop around `aceIndex` let's both declare and increment `aceIndex` instead of `aceItem`, and then at the bottom of your loop you can increment `aceIndex` at the bottem. _Technically_ you _could_ have a `aceItem++, aceIndex++`, however that's not quite as readable imo.